### PR TITLE
added rpc client to wallet

### PR DESCRIPTION
**Pull request by akekeke**
WIP

This PR:
- adds a rpc client to wallet struct
- `ListUnspent()` asks bitcoind to return UTXOs for addresses it knows about
- `NewAddress()` now also registers the newly created address  to bitcoind

### CODE REVIEW

File: internal/wallet/sign_test.go [hunk 1]
```diff
@@ -16,6 +17,10 @@ func TestWitnessSignature(t *testing.T) {
 	w, tearDownFunc := setupWallet(t)
 	defer tearDownFunc()
 
+	rpcc := &mocks.Client{}
+	rpcc = mockImportAddress(rpcc, nil)
+	w.rpc = rpcc
+
```
Comment (akekeke) [inline internal/wallet/sign_test.go]: Because `NewPubKey()` calls `ImportAddress()`, this section is added.

File: internal/wallet/utxo.go [hunk 2]
```diff
@@ -18,11 +18,17 @@ import (
 type Utxo = btcjson.ListUnspentResult
 
 // ListUnspent returns unspent transactions.
+// It asks the running bitcoind instance to return all known utxos for addresses it knows about
+func (w *wallet) ListUnspent() (utxos []Utxo, err error) {
+	return w.rpc.ListUnspent()
```
Comment (akekeke) [inline internal/wallet/utxo.go]: Renamed this function

File: internal/wallet/utxo_test.go [hunk 3]
```diff
@@ -36,25 +38,35 @@ func TestListUnspent(t *testing.T) {
...
-		fmt.Println(err)
-	}
 	assert.Nil(t, err)
+	assert.Equal(t, rutxos, utxos)
+}
+
+func mockListUnspent(c *mocks.Client, utxos []btcjson.ListUnspentResult, err error) *mocks.Client {
+	c.On("ListUnspent").Return(utxos, err)
 
+	return c
```
Comment (Jwata) [inline internal/wallet/utxo_test.go]: 👍

File: internal/wallet/wallet.go [hunk 4]
```diff
@@ -49,6 +50,10 @@ var (
 	waddrmgrNamespaceKey = []byte("waddrmgr")
 	waddrmgrKeyScope     = waddrmgr.KeyScopeBIP0084
 	wtxmgrNamespaceKey   = []byte("wtxmgr")
+
+	rpcport     = "localhost:18443" // TODO: have rpc params be read from conf file? //
+	rpcusername = "akek"
+	rpcpassword = "akek"
```
Comment (Jwata) [inline internal/wallet/wallet.go]: Yes, the RPC parameter should be defiend in a conf file.
Leave the TODO and do it later when we integrate.

File: internal/wallet/wallet.go [hunk 5]
```diff
@@ -57,11 +62,11 @@ const accountName = "dlc"
...
 	publicPassphrase []byte
-	// rpc    *rpc.BtcRPC
-	db      walletdb.DB
-	manager *waddrmgr.Manager
-	txStore *wtxmgr.Store
-	account uint32
+	rpc              rpc.Client
+	db               walletdb.DB
+	manager          *waddrmgr.Manager
+	txStore          *wtxmgr.Store
```
Comment (Jwata) [inline internal/wallet/wallet.go]: Do we still use txstore?

**Pull request merged**